extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/context/RunInSafeDuplicatedContext.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.vertx.core.runtime.context;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import jakarta.enterprise.util.Nonbinding;
+import jakarta.interceptor.InterceptorBinding;
+
+/**
+ * Check that a method in invoke on a duplicated context and mark that context as safe.
+ * This interceptor is a companion on {@link VertxContextSafetyToggle}.
+ * It means that the user knows the context is not going to be access concurrently.
+ * <p>
+ * If the method is not run on a duplicated context, the interceptor fails.
+ * If the method is invoked on an unmarked duplicated context, the context is marked as `safe` and the method is called.
+ * If the method is invoked on a `safe` duplicated context, the method is called.
+ * If the method is invoked on an `unsafe` duplicated context, the interceptor fails, except if {@link #force()} is
+ * set to {@code true}. In this case, the duplicated context is marked as `safe` and the method is called.
+ */
+@InterceptorBinding
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE, ElementType.METHOD, ElementType.CONSTRUCTOR })
+@Inherited
+public @interface RunInSafeDuplicatedContext {
```

Comment:
Who will be using this annotation? I mean the name itself is probably not very clear...

Reply:
It is mostly going to be used with virtual threads. With virtual threads, we don't know if the context is safe or not, so, we consider it unmarked. But if you need a safe context and you avoid concurrent access, you can use this annotation.